File: OLSource1/v1/VS_officedev/codesnippet/VisualBasic/walkthrough--adding-controls-to-a-document-at-run-time-in-a-vsto-add-in_4.vb
    Private Sub Application_DocumentBeforeSave(ByVal Doc As Word.Document, _
        ByRef SaveAsUI As Boolean, ByRef Cancel As Boolean) Handles Application.DocumentBeforeSave

        Dim isExtended As Boolean = Globals.Factory.HasVstoObject(Doc)


        If isExtended Then
            Dim vstoDocument As Document = Globals.Factory.GetVstoObject(Doc)


            If vstoDocument.Controls.Contains(button) Then
                vstoDocument.Controls.Remove(button)
                Globals.Ribbons.MyRibbon.addButtonCheckBox.Checked = False
            End If
        End If
    End Sub

File: OLSource1/v1/VS_officedev/codesnippet/CSharp/walkthrough--adding-controls-to-a-document-at-run-time-in-a-vsto-add-in_4.cs
        private void Application_DocumentBeforeSave(Word.Document Doc, 
            ref bool SaveAsUI, ref bool Cancel)
        {
            bool isExtended = Globals.Factory.HasVstoObject(Doc);


            if (isExtended)
            {

                Microsoft.Office.Tools.Word.Document vstoDocument = Globals.Factory.GetVstoObject(Doc);

                if (vstoDocument.Controls.Contains(button))
                {
                    vstoDocument.Controls.Remove(button);
                    Globals.Ribbons.MyRibbon.addButtonCheckBox.Checked = false;
                }
            }
        }